pkg/test/framework/components/echo/echotest/run.go

```diff
@@ -61,6 +64,28 @@ func (t *T) Run(testFn oneToOneTest) {
 	})
 }
 
+// RunFromClusters will generate and run one subtest to send traffic to
+// destination instance. This is for ingress gateway testing when source instance
+// is sending requests to destination through the ingress gateway.
+func (t *T) RunFromClusters(testFn oneClusterOneTest) {
+	t.fromEachClusterOnly(t.rootCtx, func(ctx framework.TestContext, c cluster.Cluster) {
+		t.toEachDeployment(ctx, func(ctx framework.TestContext, dstInstances echo.Instances) {
+			t.setupDst(ctx, dstInstances)
```

Comment:
This should be called for all of the `RunX` methods

Reply:
Are you suggesting to call `RunFromClusters` inside `RunToN()` and `Run()`? Those require a src instance but `RunFromClusters` don't need that. Could you elaborate more on this?